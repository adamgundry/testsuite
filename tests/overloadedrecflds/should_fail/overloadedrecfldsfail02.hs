{-# LANGUAGE OverloadedRecordFields, ExistentialQuantification, RankNTypes #-}

-- x is existential (naughty)
data T a = forall e . MkT { x :: e }

-- x and y are higher-rank
data U = MkU { x :: forall a . a -> a }
       | MkU2 { y :: (forall b . b) -> () }

-- Should generate sensible unsolved constraint errors
a = x (MkT True) :: Bool
b = x (MkU id)
c = y (MkU2 (\ _ -> ()))
d = x ((\ x -> x) :: Int -> Int) :: Bool

e :: (T Int) { foo :: t } => t
e = x (MkT True)

main = return ()
