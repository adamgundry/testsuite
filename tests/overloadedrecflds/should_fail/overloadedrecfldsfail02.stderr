
overloadedrecfldsfail02.hs:11:5:
    No instance for T {x :: Bool}
      arising from a use of the record selector ‛x’
    The field ‛x’ of ‛T’ cannot be overloaded,
      as its type is universally or existentially quantified
    In the expression: x
    In the expression: x (MkT True) :: Bool
    In an equation for ‛a’: a = x (MkT True) :: Bool

overloadedrecfldsfail02.hs:12:5:
    No instance for U {x :: t1}
      arising from a use of the record selector ‛x’
    The field ‛x’ of ‛U’ cannot be overloaded,
      as its type is universally or existentially quantified
    In the expression: x
    In the expression: x (MkU id)
    In an equation for ‛b’: b = x (MkU id)

overloadedrecfldsfail02.hs:13:5:
    No instance for U {y :: t0}
      arising from a use of the record selector ‛y’
    The field ‛y’ of ‛U’ cannot be overloaded,
      as its type is universally or existentially quantified
    In the expression: y
    In the expression: y (MkU2 (\ _ -> ()))
    In an equation for ‛c’: c = y (MkU2 (\ _ -> ()))

overloadedrecfldsfail02.hs:14:5:
    No instance for (Int -> Int) {x :: Bool}
      arising from a use of the record selector ‛x’
    The type ‛(->)’ does not have a field ‛x’
    In the expression: x
    In the expression: x ((\ x -> x) :: Int -> Int) :: Bool
    In an equation for ‛d’: d = x ((\ x -> x) :: Int -> Int) :: Bool
